### PR TITLE
[Feature] Iceberg connector supports custom catalog

### PR DESCRIPTION
**Pull request by mxdzs0612**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Related to #7647

## Problem Summary(Required) ：

Support using custom catalog in Iceberg connector.

@stephen-shelby @caneGuy @DorianZheng

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 19 / 37 (51.35%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|1|11|09.09%|[79, 80, 147, 148, 149, 150, 151, 152, 153, 154]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|4|6|66.67%|[184, 185]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|13|19|68.42%|[44, 45, 46, 47, 49, 80]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergConnector.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 18 / 47 (38.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|1|11|09.09%|[79, 80, 147, 148, 149, 150, 151, 152, 153, 154]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|2|9|22.22%|[183, 184, 185, 186, 187, 188, 189]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|14|26|53.85%|[45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 56, 88]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergConnector.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 17 / 43 (39.53%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|1|11|09.09%|[79, 80, 147, 148, 149, 150, 151, 152, 153, 154]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|2|9|22.22%|[183, 184, 185, 186, 187, 188, 189]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|13|22|59.09%|[45, 46, 47, 48, 49, 50, 51, 53, 84]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergConnector.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 17 / 43 (39.53%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|1|11|09.09%|[79, 80, 147, 148, 149, 150, 151, 152, 153, 154]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|2|9|22.22%|[182, 183, 184, 185, 186, 187, 188]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|13|22|59.09%|[45, 46, 47, 48, 49, 50, 51, 53, 84]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergConnector.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java [hunk 1]
```diff
@@ -48,7 +48,7 @@ public class IcebergTable extends Table {
 
     public static final String ICEBERG_CATALOG = "starrocks.catalog-type";
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java:49]: why not "iceberg.catalog.type"?
it is wired to use starrocks as beginning.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 17 / 43 (39.53%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|1|11|09.09%|[79, 80, 147, 148, 149, 150, 151, 152, 153, 154]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|2|9|22.22%|[182, 183, 184, 185, 186, 187, 188]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|13|22|59.09%|[45, 46, 47, 48, 49, 50, 51, 53, 84]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergConnector.java|1|1|100.00%|[]|

**Comment by mxdzs0612** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 17 / 43 (39.53%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|1|11|09.09%|[79, 80, 147, 148, 149, 150, 151, 152, 153, 154]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|2|9|22.22%|[182, 183, 184, 185, 186, 187, 188]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|13|22|59.09%|[45, 46, 47, 48, 49, 50, 51, 53, 84]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergConnector.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java (hunk 1 above)
Comment (mxdzs0612) [reply fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java:49]: I'll submit a separated PR to refractor this property after this one is merged.

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java [hunk 2]
```diff
@@ -177,13 +177,29 @@ public static Map<PartitionField, Integer> getIdentityPartitions(PartitionSpec p
...
         Map<String, String> properties = new HashMap<>();
-        properties.put(IcebergTable.ICEBERG_CATALOG, "HIVE_CATALOG");
         properties.put(IcebergTable.ICEBERG_DB, dbName);
         properties.put(IcebergTable.ICEBERG_TABLE, tblName);
+        properties.put(IcebergTable.ICEBERG_CATALOG, "HIVE_CATALOG");
         properties.put(IcebergTable.ICEBERG_METASTORE_URIS, metastoreURI);
+        return convertColumnTypes(icebergTable, properties);
+    }
+
+    private static IcebergTable convertColumnTypes(org.apache.iceberg.Table icebergTable, Map<String, String> properties)
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java:201]: convertToSRTable?

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java [hunk 3]
```diff
@@ -52,15 +71,19 @@ public Database getDb(String dbName) {
...
     }
 
     @Override
     public Table getTable(String dbName, String tblName) {
         try {
-            org.apache.iceberg.Table icebergTable = hiveCatalog.loadTable(IcebergUtil.getIcebergTableIdentifier(dbName, tblName));
-            return IcebergUtil.convertToSRTable(icebergTable, metastoreURI, dbName, tblName);
+            org.apache.iceberg.Table icebergTable
+                    = icebergCatalog.loadTable(IcebergUtil.getIcebergTableIdentifier(dbName, tblName));
+            if ("custom".equalsIgnoreCase(catalogType)) {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:83]: nit: use `IcebergCatalogType.fromString` here

**Review by DorianZheng — DISMISSED** [review]
LGTM

**Review by stephen-shelby — DISMISSED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 17 / 43 (39.53%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|1|11|09.09%|[79, 80, 147, 148, 149, 150, 151, 152, 153, 154]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|2|9|22.22%|[182, 183, 184, 185, 186, 187, 188]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|13|22|59.09%|[45, 46, 47, 48, 49, 50, 51, 53, 84]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergConnector.java|1|1|100.00%|[]|

**Review by caneGuy — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**